packages/frontend/@n8n/rest-api-client/src/utils.ts

```diff
@@ -114,6 +114,13 @@ export async function request(config: {
 		}
 
 		const errorResponseData = error.response?.data;
+		if (errorResponseData?.mfaRequired === true) {
+			console.log('Redirecting to personal settings');
+			if (window.location.pathname !== '/settings/personal') {
+				window.location.search = '';
+				window.location.pathname = '/settings/personal';
```

Comment:
I'm not sure it's possible here, but it would be better to use vue routing I think

Reply:
Yes I'm not sure if this is possible here, but I had the same thought. Maybe @cstuncsik can comment on this when he reviews this PR.

Reply:
It's possible but yes, we shoul use vue-router to do the redirection.
And the error handling should be in a custom axios interceptor

Reply:
Or for better organisation and maintainability we can have some helper function in the `@n8n/rest-api-client` frontend package

Reply:
@cstuncsik I don't understand how I can access the router from editor-ui in the `@n8n/rest-api-client` package, wouldn't this create a circle dependency?

Reply:
You could pass it, but you'd need other stuff so it's probably easier to keep this in the editor-ui package.
I'll push a commit with my recommendation